keras_core/backend/jax/distribution_lib.py

```diff
@@ -54,3 +54,45 @@ def to_jax_layout(tensor_layout):
     partition_spec = jax.sharding.PartitionSpec(*tensor_layout.axes)
     jax_mesh = to_jax_mesh(tensor_layout.device_mesh)
     return jax.sharding.NamedSharding(jax_mesh, partition_spec)
+
+
+def distribute_data(data, distribution):
+    """Distribute the input data based on the distribution setting.
+
+    Args:
+        data: `jax.Array` or a nested structure `jax.Array` to distribution
+        distribution: `keras_core.distribution.Distribution` instance. Could be
+            `None`.
+    Returns:
+        Distributed data. In the case that `distribution` is None, the original
+        data will be returned.
+    """
+    if distribution is None:
+        return data
+
+    jax_sharding = jax.tree_util.tree_map(
+        lambda d: to_jax_layout(distribution.get_data_layout(d.shape)), data
+    )
+    return jax.device_put(data, jax_sharding)
+
+
+def distribute_variable(value, variable_path, distribution):
+    """Distribute the variable init value based on the distribution setting.
+
+    Args:
+        value: `jax.Array`, the init value for distribution.
+        variable_path: str, the path of the variable, which is obtained from
+            name scope.
+        distribution: `keras_core.distribution.Distribution` instance. Could be
+            `None`.
+    Returns:
+        Distributed variable. In the case that `distribution` is None,
+        the original value will be returned.
+    """
+    if distribution is None:
+        return value
+
+    variable_sharding = to_jax_layout(
+        distribution.get_variable_layout(value.shape, variable_path)
```

Comment:
That line can be moved to `Variable._initialize`

Reply:
Done.